Classes/Controller/HaikuController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace T3docs\Examples\Controller;
+
+use T3docs\Examples\Domain\Repository\HaikuRepository;
+use TYPO3\CMS\Core\Service\FlexFormService;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+/**
+ * Class FalExampleController
+ */
+class HaikuController
+{
+    /**
+     * The back-reference to the mother cObj object set at call time
+     */
+    public ContentObjectRenderer $cObj;
+    public array $conf = [];
+    public StandaloneView $view;
+
+    public function __construct(
+        protected readonly HaikuRepository $haikuRepository,
+    ) {
+    }
+
+    /**
+     * This setter is called when the plugin is called from UserContentObject (USER)
+     * via ContentObjectRenderer->callUserFunction().
+     */
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+    public function main(string $content, array $conf): string
+    {
+        $this->conf = $conf;
+        $this->loadFlexFormSettings();
+        $this->initView();
+
+        $action = $_REQUEST['tx_examples_haiku']['action'] ?? 'list';
+        switch ($action) {
+            case 'show':
+                return $this->showAction((int)$_REQUEST['tx_examples_haiku']['haiku'] ?? 0);
+            case 'findByTitle':
+                return $this->findByTitleAction($_REQUEST['tx_examples_haiku']['haiku_title'] ?? '');
+            case 'list':
+                return $this->listAction();
+            default:
+                return $this->errorAction();
+        }
+    }
+
+    private function initView(): void
+    {
+        $this->view = GeneralUtility::makeInstance(StandaloneView::class);
+        $this->view->setLayoutRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Layouts'),
+        ]);
+        $this->view->setPartialRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Partials'),
+        ]);
+        $this->view->setTemplateRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Templates'),
+        ]);
+        $this->view->setFormat('html');
+        $this->view->setTemplate('Haiku/Default');
+        $this->view->assignMultiple([
```

Comment:
Why `assignMultiple` instead of `assign`?

Reply:
Because like this you can just add additional variables once you need them without switching between two formats. Also there is not harm in it.